install-panel.sh

```diff
@@ -749,118 +752,57 @@ function configure_nginx {
-      exit 1
-    fi
-    install_composer
-    ptdl_dl
-    create_database
-    configure
-    insert_cronjob
-    install_pteroq
-
-    if [ "$OS_VER_MAJOR" == "18" ] || [ "$OS_VER_MAJOR" == "20" ]; then
-      if [ "$CONFIGURE_LETSENCRYPT" == true ]; then
-        letsencrypt
+      if [ "$OS" == "ubuntu" ]; then
+        [ "$OS_VER_MAJOR" == "20" ] && ubuntu20_dep
+        [ "$OS_VER_MAJOR" == "18" ] && ubuntu18_dep
+      elif [ "$OS" == "debian" ]; then
+        [ "$OS_VER_MAJOR" == "9" ] && debian_stretch_dep
+        [ "$OS_VER_MAJOR" == "10" ] && debian_dep
       fi
-    fi
-  elif [ "$OS" == "debian" ]; then
-    apt_update
-    if [ "$OS_VER_MAJOR" == "9" ]; then
-      debian_stretch_dep
-    elif [ "$OS_VER_MAJOR" == "10" ]; then
-      debian_dep
-    fi
-    install_composer
-    ptdl_dl
-    create_database
-    configure
-    insert_cronjob
-    install_pteroq
-
-    if [ "$OS_VER_MAJOR" == "9" ] || [ "$OS_VER_MAJOR" == "10" ]; then
-      if [ "$CONFIGURE_LETSENCRYPT" == true ]; then
-        letsencrypt
-      fi
-    fi
-  elif [ "$OS" == "centos" ]; then
-    if [ "$OS_VER_MAJOR" == "7" ]; then
-      centos7_dep
-    elif [ "$OS_VER_MAJOR" == "8" ]; then
-      centos8_dep
-    fi
-    centos_php
-    install_composer
-    ptdl_dl
-    create_database
-    configure
-    insert_cronjob
-    install_pteroq
-    if [ "$OS_VER_MAJOR" == "7" ] || [ "$OS_VER_MAJOR" == "8" ]; then
-      if [ "$CONFIGURE_LETSENCRYPT" == true ]; then
-        letsencrypt
-      fi
-    fi
-  else
-    # exit
-    print_error "OS not supported."
-    exit 1
-  fi
-
-  # perform webserver configuration
-  if [ "$WEBSERVER" == "nginx" ]; then
-    configure_nginx
-  else
-    print_error "Invalid webserver."
-    exit 1
-  fi
-}
-
-function ask_letsencrypt {
-  if [ "$CONFIGURE_UFW" == false ] && [ "$CONFIGURE_FIREWALL_CMD" == false ]; then
-    print_warning "Let's Encrypt requires port 80/443 to be opened! You have opted out of the automatic firewall configuration; use this at your own risk (if port 80/443 is closed, the script will fail)!"
-  fi
-
-  print_warning "You cannot use Let's Encrypt with your hostname as an IP address! It must be a FQDN (e.g. panel.example.org)."
-
-  echo -e -n "* Do you want to automatically configure HTTPS using Let's Encrypt? (y/N): "
-  read -r CONFIRM_SSL
-
-  if [[ "$CONFIRM_SSL" =~ [Yy] ]]; then
-    CONFIGURE_LETSENCRYPT=true
-    ASSUME_SSL=true
-  fi
+    ;;
+
+    centos)
+      [ "$OS_VER_MAJOR" == "7" ] && yum_update
+      [ "$OS_VER_MAJOR" == "8" ] && dnf_update
+
+      [ "$CONFIGURE_FIREWALL_CMD" == true ] && firewall_firewalld
+
+      [ "$OS_VER_MAJOR" == "7" ] && centos7_dep
+      [ "$OS_VER_MAJOR" == "8" ] && centos8_dep
+    ;;
+  esac
+  
+  [ "$OS" == "centos" ] && centos_php
+  install_composer
+  ptdl_dl
+  create_database
+  configure
+  set_folder_permissions
+  insert_cronjob
+  install_pteroq
+  configure_nginx
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && letsencrypt
```

Comment:
```suggestion
  [ "$CONFIGURE_LETSENCRYPT" == true ] && letsencrypt
  true
```

Since `[ "$CONFIGURE_LETSENCRYPT" == true ] && letsencrypt` is the last command in the body of the function `main`, it's exit code defines how `main` exits. Therefore, not using Let's Encrypt would mean that the `goodbye` function never runs since the script would exit here.